src/MessageProtocol/MessageProtocolBLE.cpp

```diff
@@ -156,6 +180,24 @@ void MessageProtocolBLE::setAckFlag(bool on)
     m_ackFlag = on ? ACK_FLAG_BIT : 0x00;
 }
 
+quint32 MessageProtocolBLE::convertToQuint32(const QByteArray &data)
+{
+    QVector<quint8> l(4, 0);
+    std::transform(std::begin(data), std::end(data), std::begin(l),
+                   [](char c) {return static_cast<quint8>(c);});
+    return convertToQuint32(l[0], l[1], l[2], l[3]);
```

Comment:
I don't like this...

Data are copied in a vector just to cast to quint8. Why not just doing:

```
return convertToQuint32(static_cast<quint8>(data.at(0)), 
                        static_cast<quint8>(data.at(1)),
                        static_cast<quint8>(data.at(2)),
                        static_cast<quint8>(data.at(3)));
```

Reply:
data size can be under 4, thats why I did not implemented this way.

Reply:
Then a bunch of: ```if (data.size() < xxx)``` is way faster than allocating a QVector everytime...